src/NLog/LayoutRenderers/ExceptionLayoutRenderer.cs

```diff
@@ -411,6 +463,30 @@ protected virtual void AppendSerializeObject(StringBuilder sb, Exception ex)
             ConfigurationItemFactory.Default.ValueFormatter.FormatValue(ex, null, MessageTemplates.CaptureType.Serialize, null, sb);
         }
 
+        /// <summary>
+        /// Appends all the additional properties of an Exception like Data key-value-pairs
+        /// </summary>
+        /// <param name="sb">The <see cref="StringBuilder"/> to append the rendered data to.</param>
+        /// <param name="ex">The Exception whose properties should be appended.</param>
+        protected virtual void AppendProperties(StringBuilder sb, Exception ex)
+        {
+            string separator = string.Empty;
+            var exceptionProperties = ObjectReflectionCache.LookupObjectProperties(ex);
+            foreach (var property in exceptionProperties)
+            {
+                if (ExcludeDefaultProperties.Contains(property.Name))
+                    continue;
+
+                var propertyValue = property.Value;
+                if (propertyValue == null || string.IsNullOrEmpty(propertyValue as string))
```

Comment:
is this on purpose that we use here `as string` and we `ToString`? The latter could also return empty? (if so, please add a comment line, thanks!)

Reply:
Fixed